src/globalvars.cpp

```diff
@@ -258,19 +303,99 @@ void GlobalVars::verify_maps_found(
     const std::unordered_map<std::string, struct bpf_map *>
         &section_name_to_global_vars_map)
 {
-  for (const auto &[name, config] : global_var_map_) {
+  for (const auto &[name, config] : added_global_vars_) {
     if (!section_name_to_global_vars_map.contains(config.section)) {
       LOG(BUG) << "No map found in " << config.section
                << " which is needed to set global variable " << name;
     }
   }
 }
 
+Result<GlobalVarVals> GlobalVars::get_named_param_vals(
```

Comment:
I think you could probably isolate all the parsing state and logic and add unit tests that makes it easier to grok?